apps/src/lib/kits/maker/boards/microBit/MBFirmataWrapper.js

```diff
@@ -42,13 +39,48 @@ export default class MicrobitFirmataWrapper extends MBFirmataClient {
     } else {
       // Use the given port. Assume the port has been opened by the caller.
       // This branch is for WebSerial pathway - only opening/closing port for now.
-      this.myPort = port;
+      this.myPort = port; // port is a WebSerialPortWrapper
+      this.serialPortWebSerial = port.port;
       this.myPort.on('data', this.dataReceived.bind(this));
+      this.requestFirmataVersion();
+      this.requestFirmwareVersion();
 
       return Promise.resolve();
     }
   }
 
+  // Create and return a copy of this MBFirmataClient with references to the WebSerialPortWrapper removed.
+  getBoardClientWithoutPort() {
```

Comment:
Nit: we should add a comment about why we are doing this since it's a bit counter-intuitive. :)